lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2073,39 +2076,43 @@ def deploy_vm(self):
             self.folder = '/%(folder)s' % self.params
         self.folder = self.folder.rstrip('/')
 
-        datacenter = self.cache.find_obj(self.content, [vim.Datacenter], self.params['datacenter'])
-        if datacenter is None:
+        # get all datacenters with the same name
+        datacenter_list = find_obj(self.content, [vim.Datacenter], self.params['datacenter'], first=False)
+        if len(datacenter_list) == 0:
             self.module.fail_json(msg='No datacenter named %(datacenter)s was found' % self.params)
-
-        dcpath = compile_folder_path_for_object(datacenter)
-
-        # Nested folder does not have trailing /
-        if not dcpath.endswith('/'):
-            dcpath += '/'
-
-        # Check for full path first in case it was already supplied
-        if (self.folder.startswith(dcpath + self.params['datacenter'] + '/vm') or
-                self.folder.startswith(dcpath + '/' + self.params['datacenter'] + '/vm')):
-            fullpath = self.folder
-        elif self.folder.startswith('/vm/') or self.folder == '/vm':
-            fullpath = "%s%s%s" % (dcpath, self.params['datacenter'], self.folder)
-        elif self.folder.startswith('/'):
-            fullpath = "%s%s/vm%s" % (dcpath, self.params['datacenter'], self.folder)
         else:
-            fullpath = "%s%s/vm/%s" % (dcpath, self.params['datacenter'], self.folder)
-
-        f_obj = self.content.searchIndex.FindByInventoryPath(fullpath)
-
+            dcpath = []
```

Comment:
`dcpath` probably got to the incorrect context scope

Reply:
Do you mean "else" here is no need?

Reply:
Sorry, hadn't seen this else here, and it's fine to have it like now.